src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -1007,9 +1007,24 @@ public BoundExpression Convert(TypeSymbol type, BoundExpression arg, ConversionK
             { WasCompilerGenerated = true };
         }
 
-        public BoundExpression Array(TypeSymbol elementType, BoundExpression[] elements)
+        public BoundExpression ArrayOrEmpty(TypeSymbol elementType, BoundExpression[] elements)
         {
-            return Array(elementType, elements.AsImmutableOrNull());
+            return ArrayOrEmpty(elementType, elements.AsImmutableOrNull());
+        }
+
+        public BoundExpression ArrayOrEmpty(TypeSymbol elementType, ImmutableArray<BoundExpression> elements)
+        {
+            if (elements.Length == 0)
```

Comment:
Yes. And from AsImmutableOrNull above it seems like it is possible. Will have to fix this. Thanks!